a lot of providers documentation (#4287)
centreon/centreon-documentation: cloud/alerts-notifications/ticketing/providers/glpi-api-rest.md

---
id: glpi-api-rest
title: GlpiApiRest
---

## Features information

| open ticket | close ticket (from Centreon to Glpi) | Handle custom fields |
| -- | -- | -- |
| ✓ | ✓ | ✘ |

## Prerequisites

### Network Flow

| Source | Destination | Protocol/Port |
| -- | -- | -- |
| Centreon central server | Glpi | TCP/443 (https) or TCP/80 (http) |

### Account

You need the following information

- Api token (also known as app token)
- User token
- Serveur address

> The aformentioned account must be able to at least be able to open a ticket through the **/Ticket API endpoint** and to authenticate itself through the **/initSession API endpoint**.

Reviewer: ```suggestion
The aforementioned account must at least be able to open a ticket through the **/Ticket API endpoint** and to authenticate itself through the **/initSession API endpoint**.
```

The connector will also try to access the following API endpoints depending on the configuration of your open ticket rule.

- /User
- /Group

> - /getMyEtities

Reviewer: ```suggestion
- /getMyEntities
```

- /Supplier
- /itilCategory

> Some tests commands that you can run from your Centreon central server are available in the [Test commands chapter](#test-commands)

Reviewer: ```suggestion
Some test commands that you can run from your Centreon central server are available in the [Test commands chapter](#test-commands)
```

## Retrieved data

> This open ticket connector can retrieve the following informations from your Glpi server

Reviewer: ```suggestion
This open ticket connector can retrieve the following information from your Glpi server
```

- Entities
- Users
- Groups
- Suppliers
- ITIL categories

If you need more information regarding retrieved data from an open ticket connector, please read the [retrieved data chapter](../mapping.md) from the Open Ticket module global documentation.

## Additionnal data

This open ticket connector can also send the following information when opening a ticket

- Urgency
- Impact
- Priority
- Group and User role

Those information are not retrieved from Glpi. They are configured in your open ticket rule as [custom lists](../mapping.md#custom-list-definition)

## Test commands

> The below Curl commands must be run from your central server. You need to replace everything between `<>` for example `<glpi_address>` may become **my_glpi.local**

Reviewer: ```suggestion
The Curl commands listed below must be run from your central server. You need to replace everything between `<>` for example `<glpi_address>` may be replaced with **my_glpi.local**
```

```

### Get Glpi users

```bash
curl --location 'https://<glpi_address>/<glpi_api_path>/User' \
--header 'Authorization: Bearer <authentication_token>' \
--header 'Content-Type: application/json' \
--header 'App-Token: <app_token>'  \
--header 'Session-Token: <session_token>'
```

### Get Glpi groups

```bash
curl --location 'https://<glpi_address>/<glpi_api_path>/Group' \
--header 'Authorization: Bearer <authentication_token>' \
--header 'Content-Type: application/json' \
--header 'App-Token: <app_token>'  \
--header 'Session-Token: <session_token>'
```

### Get Glpi suppliers

```bash
curl --location 'https://<glpi_address>/<glpi_api_path>/Supplier' \
--header 'Authorization: Bearer <authentication_token>' \
--header 'Content-Type: application/json' \
--header 'App-Token: <app_token>'  \
--header 'Session-Token: <session_token>'
```

### Get Glpi ITIl categories

```bash
curl --location 'https://<glpi_address>/<glpi_api_path>/itilCategory' \
--header 'Authorization: Bearer <authentication_token>' \
--header 'Content-Type: application/json' \
--header 'App-Token: <app_token>'  \
--header 'Session-Token: <session_token>'
```

### Open a ticket

> Keep in mind that the data in the below command is just an example and your Glpi server may ask you to add mandatory data

Reviewer: ```suggestion
Keep in mind that the data in the command down below is just an example and your Glpi server may ask you to add mandatory data
```
